save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPluginConfig.kt

```diff
@@ -31,41 +31,72 @@ import kotlinx.serialization.UseSerializers
  */
 @Serializable
 data class WarnPluginConfig(
-    val execFlags: String,
-    val warningsInputPattern: Regex,
-    val warningsOutputPattern: Regex,
+    val execFlags: String? = null,
+    val warningsInputPattern: Regex? = null,
+    val warningsOutputPattern: Regex? = null,
     val warningTextHasLine: Boolean? = null,
     val warningTextHasColumn: Boolean? = null,
-    val lineCaptureGroup: Int?,
-    val columnCaptureGroup: Int?,
-    val messageCaptureGroup: Int,
+    val lineCaptureGroup: Int? = null,
+    val columnCaptureGroup: Int? = null,
+    val messageCaptureGroup: Int? = null,
     val exactWarningsMatch: Boolean? = null,
     val testNameSuffix: String? = null,
 ) : PluginConfig {
     override val type = TestConfigSections.WARN
+    private val testName: String = testNameSuffix ?: "Test"
 
     /**
      *  @property resourceNamePattern regex for the name of the test files.
      */
-    val resourceNamePattern: Regex = resourceNamePattern()
+    val resourceNamePattern: Regex = Regex("""(.+)${(testName)}\.[\w\d]+""")
 
     override fun mergeWith(otherConfig: PluginConfig): PluginConfig {
         val other = otherConfig as WarnPluginConfig
         return WarnPluginConfig(
             this.execFlags ?: other.execFlags,
-            this.warningsInputPattern,
-            this.warningsOutputPattern,
+            this.warningsInputPattern ?: other.warningsInputPattern,
+            this.warningsOutputPattern ?: other.warningsOutputPattern,
             this.warningTextHasLine ?: other.warningTextHasLine,
             this.warningTextHasColumn ?: other.warningTextHasColumn,
             this.lineCaptureGroup ?: other.lineCaptureGroup,
             this.columnCaptureGroup ?: other.columnCaptureGroup,
-            this.messageCaptureGroup,
+            this.messageCaptureGroup ?: other.columnCaptureGroup,
             this.exactWarningsMatch ?: other.exactWarningsMatch,
             this.testNameSuffix ?: other.testNameSuffix
```

Comment:
don't see a default for testNameSuffix

Reply:
We have separate variables `testName` for `warn` `resourceNameTest` `resourceNameExpected` which are already who perform this role, and I use them as arguments in validation

The reason, why we still need such variables - we use them for resource detection, which is doing before validation